dev/com.ibm.ws.ejbcontainer/src/com/ibm/ws/ejbcontainer/osgi/internal/EJBMDOrchestratorImpl.java

```diff
@@ -1342,32 +1344,43 @@ protected void processZOSMetadata(BeanMetaData bmd) {
         if (isTraceOn && tc.isEntryEnabled())
             Tr.entry(tc, "processZOSMetadata : " + bmd);
 
-        if (EJSPlatformHelper.isZOS()) {
-            bmd.m_syncToOSThreadValue = getSyncToOSThreadSetting(ContainerConfigConstants.syncToOSThreadSetting, bmd);
+        if (isZOS()) {
+            // Call the BeanMetaData's initializeSyncToOSThread method to check environment entry
+            bmd.initializeSyncToOSThread();
+            
+            // If not set by environment entry, check the binding
+            if (!bmd.m_syncToOSThreadValue) {
+                bmd.m_syncToOSThreadValue = getSyncToOSThreadSetting(ContainerConfigConstants.syncToOSThreadSetting, bmd);
```

Comment:
I think there is a misunderstnding what `getSyncToOSThreadSetting` is doing.  That method already does what is needed.
1. it obtains the value from the env entry
2 . if not set, then it looks for a binding override.

When a java:comp lookup is performed, that will be re-directed to this very class, and it will look at:
```
bmd.ivJavaColonCompEnvMap.get(requiredAttr)
```
that is the map that is the very underlying data behind java:comp.  It is far more efficient to just call `getSyncToOSThreadSetting` and have it look in the map directly, rather than calling naming only to be re-directed back here.